exec/java-exec/src/test/java/org/apache/drill/TestUnionAll.java

```diff
@@ -634,10 +634,14 @@ public void testFilterPushDownOverUnionAll() throws Exception {
         + "order by n_regionkey";
 
     // Validate the plan
-    final String[] expectedPlan = {".*Filter.*\n" +
-            ".*UnionAll.*\n" +
-            ".*Scan.*columns=\\[`n_regionkey`\\].*\n" +
-            ".*Scan.*columns=\\[`r_regionkey`\\].*"};
+    final String[] expectedPlan = {"Sort.*\n" +
```

Comment:
Apart from these tests, do we have any unit tests that produce empty results once the filter is pushed on either side of the union-all ?  I expect that there could be some issues uncovered due to empty batch handling.  However, those issues could be treated as separate JIRA.

Reply:
Looks like there are no issues related to it. I have added additional test case to verify it:
`select n_nationkey from (select n_nationkey, n_name, n_comment from cp."tpch/nation.parquet" union all select r_regionkey, r_name, r_comment  from cp."tpch/region.parquet") where n_nationkey > 4;`
When filter is pushed down the result from right side of UNION will be empty.

Generally speaking using UNION operator with empty data was resolved in [DRILL-4185](https://issues.apache.org/jira/browse/DRILL-4185) and we have test cases with empty data batches.